internal/expr/map.go

```diff
@@ -0,0 +1,54 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expr
+
+import (
+	"fmt"
+)
+
+// ExpressionMap is a map of expressions.
+type ExpressionMap struct {
+	expressions map[string]*Expression
+}
+
+// Extract extracts a value map from the record using the expression map.
+func (e *ExpressionMap) Extract(record Record) map[string]any {
+	results := map[string]any{}
+	for key, expression := range e.expressions {
+		value, err := expression.Evaluate(record)
+		if err != nil || value == nil {
+			continue
```

Comment:
should errors be treated the same as non matching expressions? should we have entries for non matching expressions in the results (i haven't actually looked at the code yet where this is used...more of a general question only about this function)?

Reply:
There's a distinction between matching records and extracting attributes. Even if a record is matched, it doesn't mean that all of the attributes the user defines will exist on that log record. In this case, if the user defines the location of one of their attributes as `body.embedded.field` and the `embedded` field doesn't exist, the evaluate expression will return an error. This shouldn't cause the rest of the attribute extraction to fail. Instead, it just means that attribute doesn't exist in the way the user defined and should be omitted.

Reply:
@jmwilliams89 in case it wasn't clear, i wasn't suggesting that no matches here should error. i just noticed that errors and no match were treated the same in that they weren't added to the results and was wondering if it was worth having a distinction (even if both should not cause a problem). sounds like no so i'll resolve